v2/internal/controllers/edge_case_test.go

```diff
@@ -221,3 +176,69 @@ func Test_CreateStorageAccountWithoutRequiredProperties_Rejected(t *testing.T) {
 	// Create it again
 	tc.CreateResourcesAndWait(acctCopy)
 }
+
+func Test_AzureNameImmutability(t *testing.T) {
+	t.Parallel()
+
+	tc := globalTestContext.ForTest(t)
+
+	rg := tc.CreateTestResourceGroupAndWait()
+
+	acct := createStorageAccount(tc, rg)
+	tc.CreateResourcesAndWait(acct)
+
+	// Patch the account to change AzureName
+	newAzureName := "test123"
+	old := acct.DeepCopy()
+	acct.Spec.AzureName = newAzureName
+	tc.PatchAndExpectError(old, acct)
```

Comment:
Maybe instead of doing `PatchAndExpectError` we should just have a `Patch` that returns the error. Then you can actually assert that it has the right shape as well?

Reply:
We already have `Patch` which expects error to be nil. Can make this one return `error` and then assert on if error is not nil

Reply:
Totally agree that's the right approach